pkgs/development/tools/mongosh/default.nix

```diff
@@ -1,37 +1,78 @@
 { lib
+, stdenv
+, fetchFromGitHub
 , buildNpmPackage
-, fetchurl
+, git
+, python3
+, darwin
+, krb5
+, libmongocrypt
 , testers
 , mongosh
 }:
 
-let
-  source = lib.importJSON ./source.json;
-in
-buildNpmPackage {
+buildNpmPackage rec {
   pname = "mongosh";
-  inherit (source) version;
+  version = "2.0.1";
 
-  src = fetchurl {
-    url = "https://registry.npmjs.org/mongosh/-/${source.filename}";
-    hash = source.integrity;
+  src = fetchFromGitHub {
+    owner = "mongodb-js";
+    repo = "mongosh";
+    rev = "v${version}";
+    hash = "sha256-W8NGoT/kNFjEElQraEg96OqlNtESsdD8WP+aWp6autg=";
   };
 
+  # lerna ERR! ENOGIT The git binary was not found, or this is not a git repository.
   postPatch = ''
-    ln -s ${./package-lock.json} package-lock.json
+    if [ -x "$(command -v git)" ]; then
+      export HOME="$(mktemp -d)"
+      git config --global user.name "Nix Builder"
+      git config --global user.email "nix-builder@nixos.org"
+      git config --global init.defaultBranch "main"
+      git init .
+    fi
```

Comment:
Wait is this all that is needed to make lerna "work" here?

Reply:
I'd also like to drop the constraint that the source must be a Git repo; I didn't manage to do that.

Reply:
I wonder where that happens. If you really got lerna to work with `buildNpmPackage`, though, with such a minor change, then it bodes well for other packages to be able to do the same without too much effort (I'd mentally written lerna off as a would-be-nice-to-have-in-the-future-but-not-priority thing due to pains with it before)